### PR TITLE
fix(json): make sure to parse a list of dict for_sample

### PR DESCRIPTION
**Pull request by aarnphm**

### CODE REVIEW

File: src/bentoml/_internal/io_descriptors/json.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def openapi_example(self):
                 self.sample
             ):
                 return self.sample.dict()
-            elif isinstance(self.sample, str):
+            elif isinstance(self.sample, (str, list)):
```
Comment (mqk) [inline src/bentoml/_internal/io_descriptors/json.py:283]: Shouldn't this addition of `list` happen further down on l. 292? We want to support `list` in addition to `dict`, not in addition to `str`.

Comment (aarnphm) [reply src/bentoml/_internal/io_descriptors/json.py:283]: You have to dump the list with JSON. This line ensures list of a complex datatype can be serialized.

**Review by bojiang — APPROVED** [review]

File: src/bentoml/_internal/io_descriptors/json.py (hunk 1 above)
Comment (bojiang) [reply src/bentoml/_internal/io_descriptors/json.py:283]: @aarnphm Could you provide some examples here?

Comment (aarnphm) [reply src/bentoml/_internal/io_descriptors/json.py:283]: similar to the one that @mqk provides:
```python
@svc.api(input=JSON.from_sample([{"a": 1, "b": 2}, {"a": 2, "b": 4}]), output=JSON())
```

**Pull request merged**